baker/archival/ArchivalBaker.ts

```diff
@@ -0,0 +1,276 @@
+        const imageMetadataDictionary = await getAllImages(trx).then((images) =>
+            keyBy(images, "filename")
+        )
+
+        const grapherIds = grapherChecksumsObjsToBeArchived.map(
+            (c) => c.chartId
+        )
+        const grapherConfigs = await db
+            .knexRaw<{
+                chartId: number
+                chartConfigId: string
+                fullConfig: JsonString
+            }>(
+                trx,
+                `-- sql
+            SELECT c.id AS chartId, cc.id AS chartConfigId, cc.full AS fullConfig
+            FROM charts c
+            JOIN chart_configs cc ON cc.id = c.configId
+            WHERE c.id IN (?)
+        `,
+                [grapherIds]
+            )
+            .then((rows) =>
+                rows.map((r) => ({
+                    chartId: r.chartId,
+                    chartConfigId: r.chartConfigId,
+                    config: parseChartConfig(r.fullConfig),
+                }))
+            )
+        const allVariableIds = new Set(
+            grapherConfigs.flatMap(
+                (c) => c.config.dimensions?.map((d) => d.variableId) ?? []
+            )
+        )
+        const variableFiles = await archiveVariableIds(
+            [...allVariableIds],
+            archiveDir
+        )
+
+        let i = 0
+        for (const { chartId, chartConfigId, config } of grapherConfigs) {
+            i++
+            const runtimeFiles = { ...commonRuntimeFiles }
+
+            for (const dim of config.dimensions ?? []) {
+                if (dim.variableId) {
+                    const variableId = dim.variableId
+                    Object.assign(runtimeFiles, variableFiles[variableId])
+                }
+            }
+
+            const checksumsObj = grapherChecksumsObjsToBeArchived.find(
+                (c) => c.chartId === chartId
+            )
+            if (!checksumsObj)
+                throw new Error(
+                    `Could not find checksums for chartId ${chartId}, this shouldn't happen`
+                )
+
+            const manifest = await assembleManifest({
+                staticAssetMap,
+                runtimeAssetMap: runtimeFiles,
+                checksumsObj,
+                archivalDate: date.formattedDate,
+                chartConfigId,
+            })
+            await bakeSingleGrapherPageForArchival(dir, config, trx, {
+                imageMetadataDictionary,
+                staticAssetMap,
+                runtimeAssetMap: runtimeFiles,
+                manifest,
+            })
+            manifests[chartId] = manifest
+
+            console.log(`${i}/${grapherConfigs.length} ${config.slug}`)
+        }
+        console.log(`Baked ${grapherConfigs.length} grapher pages`)
+    })
+
+    if (copyToLatestDir) {
+        const latestDir = path.join(archiveDir, "latest")
+        await fs.remove(latestDir)
+        await fs.copy(dir, latestDir)
+        console.log(`Copied ${dir} to ${latestDir}`)
+    }
```

Comment:
Note to self: Currently, this just makes `latest` be the exact same as the latest timestamp folder.
We'll want to change this to be cumulative: Copy all files contained in `[latestDate]` over, but don't delete any of the others already contained in `latest`.